lib/model/model.go

```diff
@@ -1539,13 +1540,121 @@ func sendIndexTo(minSequence int64, conn protocol.Connection, folder string, fs
 }
 
 func (m *Model) updateLocalsFromScanning(folder string, fs []protocol.FileInfo) {
-	m.updateLocals(folder, fs)
+	m.fmut.RLock()
+	folderCfg := m.folderCfgs[folder]
+	m.fmut.RUnlock()
+
+	folderrunner := m.folderRunners[folder]
+	var newfs []protocol.FileInfo
+
+	// if this folder is a "write/pull only" folder, then we'll have to undo any local changes
+	if folderCfg.Type == config.FolderTypeWriteOnly {
+		// reject and undo all local changes
+		// GAP: need to handle file renaming also (see rwfolder line 511+ comparing blocks)
+
+		fileDeletions := []protocol.FileInfo{}
+		dirDeletions := []protocol.FileInfo{}
+
+		for _, file := range fs {
+			if strings.Contains(file.Name, ".sync-conflict-") {
+				// this is a conflict copy, let's move on to the next file
+				continue
+			}
+
+			objType := "file"
+			action := "modified"
+			correctiveaction := "resync"
+
+			if file.IsDirectory() {
+				objType = "dir"
+			}
+
+			if file.IsDeleted() {
+				action = "deleted"
+			}
 
+			if len(file.Version.Counters) == 1 && file.Version.Counters[0].Value == 1 {
+				// A file, directory or symlink was added, which we'll have to remove again
+				action = "added"
+				if folderCfg.DeleteLocalChanges {
+					correctiveaction = "deleted"
+					if file.IsDirectory() {
+						dirDeletions = append(dirDeletions, file)
+					} else {
+						fileDeletions = append(fileDeletions, file)
+					}
+				} else {
+					correctiveaction = "none"
```

Comment:
Agreed, that could cause quite a bit of unnecessary load on the system.
I'm currently trying to understand how this all ties together. Do you have any suggestions on what the best place to catch these is?